File: app/src/main/java/com/example/android/todonotesapp/LoginActivity.java
package com.example.android.todonotesapp;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Toast;

public class LoginActivity extends AppCompatActivity {

    EditText editTextFullName, editTextUserName;
    Button buttonLogin;
    SharedPreferences sharedPreferences;
    SharedPreferences.Editor editor;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate( savedInstanceState );
        setContentView( R.layout.activity_login );

        editTextFullName = findViewById( R.id.editTextFullName );
        editTextUserName = findViewById( R.id.editTextUserName );
        buttonLogin = findViewById( R.id.buttonLogin );

        setupSharedPreferences();

        View.OnClickListener clickAction = new View.OnClickListener() {

            @Override
            public void onClick(View view) {

                String fullName = editTextFullName.getText().toString();
                String userName = editTextUserName.getText().toString();

                if (!TextUtils.isEmpty( fullName ) && !TextUtils.isEmpty( userName )) {
                    Intent intent = new Intent( LoginActivity.this, MyNotesActivity.class );
                    intent.putExtra( AppConstant.FULL_NAME, fullName );
                    startActivity( intent );
                    saveLoginStatus();
                    saveFullName(fullName);
                }else {
                    Toast.makeText( LoginActivity.this,"Fields can't be empty", Toast.LENGTH_SHORT ).show();
                }
            }

        };
        buttonLogin.setOnClickListener(clickAction);
}
    private void saveLoginStatus() {
        editor = sharedPreferences.edit();
        editor.putBoolean( PrefConstant.IS_LOGGED_IN, true );
        editor.apply();
    }

    private void saveFullName(String fullName) {
        editor = sharedPreferences.edit();
        editor.putString( PrefConstant.FULL_NAME, fullName );
        editor.apply();
    }

    private void setupSharedPreferences() {
        sharedPreferences = getSharedPreferences( PrefConstant.SHARED_PREFERENCE_NAME,MODE_PRIVATE );
    }
    }

File: app/src/main/java/com/example/android/todonotesapp/LoginActivity.kt
package com.example.android.todonotesapp

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class LoginActivity :AppCompatActivity() {
    lateinit var editTextFullName: EditText
    lateinit var editTextUserName: EditText
    lateinit var buttonLogin: Button
    lateinit var sharedPreferences: SharedPreferences
    lateinit var editor: SharedPreferences.Editor

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)
        bindViews()
        setupSharedPreferences()
    }

    private fun bindViews() {
        editTextFullName = findViewById(R.id.editTextFullName)
        editTextUserName = findViewById(R.id.editTextUserName)
        buttonLogin = findViewById(R.id.buttonLogin)
        val clickAction = object : View.OnClickListener{
            override fun onClick(v: View?) {
                val fullName = editTextFullName.text.toString()
                val userName = editTextUserName.text.toString()
                if(fullName.isNotEmpty() && userName.isNotEmpty()){
                    val intent = Intent(this@LoginActivity, MyNotesActivity::class.java)
                    intent.putExtra(AppConstant.FULL_NAME,fullName)
                    startActivity(intent)
                    saveFullName(fullName)
                    saveLoginStatus()
                }else{

                }
            }

        }

        buttonLogin.setOnClickListener(clickAction)
    }

    private fun saveLoginStatus() {
        editor = sharedPreferences.edit()
        editor.putBoolean(PrefConstant.IS_LOGGED_IN, true)
        editor.apply()

    }

    private fun saveFullName(fullName: String) {
        editor = sharedPreferences.edit()
        editor.putString(PrefConstant.FULL_NAME, fullName)
        editor.apply()

    }

    private fun setupSharedPreferences() {
        sharedPreferences = getSharedPreferences(PrefConstant.SHARED_PREFERENCE_NAME, Context.MODE_PRIVATE)

    }
}